src/window_main/components/misc/Slider.tsx

```diff
@@ -52,13 +74,21 @@ export default function Slider(props: SliderProps): JSX.Element {
         onInput={handleOnInput}
       ></input>
       <div className="slider_marks_container_hor">
-        {" "
-          .repeat(stepsNumber + 1)
-          .split("")
-          // just a hack to get an array of length N to map as marks
-          .map((c: string, i: number) => {
-            return <div key={i} className="slider_mark_hor" />;
-          })}
+        {posArray.map((c: SliderPosition, i: number) => {
+          return (
+            <div className="slider_mark_outer" key={c.text + "-" + i}>
+              <div
+                className="slider_mark_hor"
+                style={{ backgroundColor: c.color, opacity: c.hide ? 0 : 1 }}
+              />
+              {c.text !== "" ? (
+                <div className="slider_mark_text">{c.text}</div>
+              ) : (
+                <></>
+              )}
```

Comment:
Nit: remove "empty" final clause in ternary by just using boolean `&&`:
```suggestion
              {c.text !== "" && (
                <div className="slider_mark_text">{c.text}</div>
              )}
```